src/sys/socket/mod.rs

```diff
@@ -987,6 +1013,30 @@ impl ControlMessageOwned {
                 let content_type = unsafe { ptr::read_unaligned(p as *const u8) };
                 ControlMessageOwned::TlsGetRecordType(content_type.into())
             },
+            #[cfg(any(linux_android, target_os = "freebsd"))]
+            #[cfg(feature = "net")]
+            (libc::IPPROTO_IP, libc::IP_TTL) => {
+                let ttl: u8 = unsafe { ptr::read_unaligned(p as *const u8) };
+                ControlMessageOwned::Ipv4RecvTtl(ttl)
+            },
+            #[cfg(any(linux_android, target_os = "freebsd"))]
+            #[cfg(feature = "net")]
+            (libc::IPPROTO_IPV6, libc::IPV6_HOPLIMIT) => {
+                let ttl: u8 = unsafe { ptr::read_unaligned(p as *const u8) };
+                ControlMessageOwned::Ipv6RecvHopLimit(ttl)
+            },
+            #[cfg(any(linux_android, target_os = "freebsd"))]
+            #[cfg(feature = "net")]
+            (libc::IPPROTO_IP, libc::IP_TOS) => {
```

Comment:
Same here, FreeBSD uses `IP_RECVTOS`

Reply:
I am not 100% sure I follow here. Is the suggestion to have 2 different blocks, one for Linux and one for FreeBSD?

Reply:
That's right, sorry for being unclear.

Reply:
Done.